### PR TITLE
Experiment configuration system

### PR DESCRIPTION
**Pull request by AmanDaVinci**
Adds hydra configuration system for running and recording experiments.

### CODE REVIEW

File: dqn.py [hunk 1]
```diff
@@ -150,48 +154,56 @@ def save_checkpoint(self, name):
...
+cs.store(name="config", node=RLConfig)
+log = logging.getLogger(__name__)
 
-    args = parser.parse_args()
+@hydra.main(config_name="config")
+def main(config: RLConfig) -> None:
+    """ Runs a training experiment based on the given hydra configuration """
 
-    env = gym.envs.make(args.env)
+    print(f"Launched! Experiment logs available at {os.getcwd()}.")
```
Comment (edudev) [inline dqn.py:181]: Is it really at the cwd? Does hydra change the cwd?

**Review by edudev — APPROVED** [review]
LGTM

File: dqn.py (hunk 1 above)
Comment (AmanDaVinci) [reply dqn.py:181]: yeah, each run gets its own "experiment" directory. This way we can do multiple runs in parallel (in principle) for hyperparameter sweeps.

Comment (edudev) [reply dqn.py:181]: Yeah, that's what I'd expect. I was just surprised that it changes the cwd.

**Comment by VaniOFX** [comment]
Interesting! Merged ;)

**Pull request merged**